statemachine/src/main/java/com/amazonaws/stepfunctions/cloudformation/statemachine/DeleteHandler.java

```diff
@@ -33,9 +33,7 @@
 
             if (!currentContext.isDeletionStarted()) {
                 if (!doesStateMachineExist(model, proxy, sfnClient)) {
-                    return ProgressEvent.<ResourceModel, CallbackContext>builder()
-                            .status(OperationStatus.SUCCESS)
-                            .build();
+                    throw getStateMachineDoesNotExistException();
```

Comment:
although we're reverting here - it is now a breaking change right? users who are currently observing success in this scenario will start to see an exception. 

we should clearly document that

Reply:
sync'd offline - this change had not been released to any users so it is not breaking to any users.
resolving this comment.